webpack/scenes/ContentViews/Details/Repositories/ContentViewRepositories.js

```diff
@@ -0,0 +1,155 @@
+  const response = useSelector(state => selectCVRepos(state, cvId), shallowEqual);
+  const status = useSelector(state => selectCVReposStatus(state, cvId), shallowEqual);
+  const error = useSelector(state => selectCVReposError(state, cvId), shallowEqual);
+  const baseTypeState = {};
+  Object.keys(repoTypes).forEach((type) => {
+    baseTypeState[type] = true;
+  });
+
+  const [rows, setRows] = useState([]);
+  const [metadata, setMetadata] = useState({});
+  const [typeChecked, setTypeChecked] = useState(baseTypeState);
+  const [statusChecked, setStatusChecked] = useState({ [ADDED]: true, [NOT_ADDED]: true });
+
+  const columnHeaders = [
+    { title: __('Type'), transforms: [fitContent] },
+    __('Name'), __('Product'), __('Sync state'), __('Content'),
+    { title: __('Status') },
+  ];
+  const loading = status === STATUS.PENDING;
+
+  const buildRows = (results) => {
+    const newRows = [];
+    results.forEach((repo) => {
+      const {
+        id,
+        content_type: contentType,
+        name,
+        added_to_content_view: addedToCV,
+        product: { id: productId, name: productName },
+        content_counts: counts,
+        last_sync_words: lastSyncWords,
+        last_sync: lastSync,
+      } = repo;
+
+      const cells = [
+        { title: <Bullseye><RepoIcon type={contentType} /></Bullseye> },
+        { title: <Link to={urlBuilder(`products/${productId}/repositories`, '', id)}>{name}</Link> },
+        productName,
+        { title: <LastSync {...{ lastSyncWords, lastSync }} /> },
+        { title: <ContentCounts {...{ counts, productId }} repoId={id} /> },
+        {
+          title: <RepoAddedStatus
+            added={addedToCV || (statusChecked[ADDED] && !statusChecked[NOT_ADDED])}
+          />,
+        },
+      ];
+
+      newRows.push({ cells });
+    });
+    return newRows;
+  };
+
+  const onSelect = (_event, isSelected, rowId) => {
+    let newRows;
+    if (rowId === -1) {
+      newRows = rows.map(row => ({ ...row, selected: isSelected }));
+    } else {
+      newRows = [...rows];
+      newRows[rowId].selected = isSelected;
+    }
+
+    setRows(newRows);
+  };
+
+  const getCVReposWithOptions = (params = {}) => {
+    const contentTypeParam = [];
+    Object.keys(typeChecked).forEach((type) => {
+      if (typeChecked[type]) contentTypeParam.push(repoTypes[type]);
+    });
+    const allParams = { ...params, content_type: contentTypeParam };
+    return getContentViewRepositories(cvId, allParams, statusChecked);
+  };
+
+  useEffect(() => {
+    dispatch(getCVReposWithOptions());
+  }, [statusChecked, typeChecked]);
+
+  useEffect(() => {
+    const { results, ...meta } = response;
+    setMetadata(meta);
+
+    if (!loading && results) {
+      const newRows = buildRows(results);
+      setRows(newRows);
+    }
+  }, [JSON.stringify(response)]);
+
+  useEffect(() => {
+    if (!statusChecked[ADDED] && !statusChecked[NOT_ADDED]) setStatusChecked({ [ADDED]: true });
```

Comment:
If only Added is checked, and I want to see just Not Added, this results in my click not doing anything.  The user would think it's broken or there's something wrong.

I think this logic should be changed to "if a checkbox is the only one checked and the user tries to uncheck it, check the other box."